src/k8s/pkg/k8sd/app/hooks_bootstrap.go

```diff
@@ -28,20 +28,25 @@ import (
 // onBootstrap configures local services then writes the cluster config on the database.
 func (a *App) onBootstrap(s *state.State, initConfig map[string]string) error {
 	if workerToken, ok := initConfig["workerToken"]; ok {
-		return a.onBootstrapWorkerNode(s, workerToken)
+		return a.onBootstrapWorkerNode(s, workerToken, initConfig)
 	}
 
 	return a.onBootstrapControlPlane(s, initConfig)
 }
 
-func (a *App) onBootstrapWorkerNode(s *state.State, encodedToken string) error {
+func (a *App) onBootstrapWorkerNode(s *state.State, encodedToken string, initConfig map[string]string) error {
```

Comment:
this starts to look like high coupling. Let's only deal with `initConfig` on the `onBootstrap`, then pass individual types to `onBootstrapWorkerNode` and `onBootstrapControlPlane`.